src/Microsoft.TemplateEngine.Cli/TemplateResolution/ListOrHelpTemplateListResolutionResult.cs

```diff
@@ -0,0 +1,241 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.TemplateEngine.Edge.Template;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.TemplateEngine.Utils;
+using System.Collections.ObjectModel;
+using Microsoft.TemplateEngine.Abstractions;
+
+namespace Microsoft.TemplateEngine.Cli.TemplateResolution
+{
+    /// <summary>
+    /// Class represents the template list resolution result to be used for listing or displaying help purposes
+    /// </summary>
+    public class ListOrHelpTemplateListResolutionResult
+    {
+        public ListOrHelpTemplateListResolutionResult(string templateName, IReadOnlyCollection<ITemplateMatchInfo> coreMatchedTemplates)
+        {
+            _templateName = templateName;
```

Comment:
if templateName was null or empty string, would it make sense to detect & handle it here in the .ctor?

Reply:
In fact template name is currently not used, I will remove it from the class. It make sense to consider injecting the whole commandInput to template resolution result, however it is not considered in this bulk of refactoring.